src/parser/jobs/sch/modules/Aetherflow.tsx

```diff
@@ -0,0 +1,258 @@
+						<Table.Cell>
+							<Grid>
+								{aetherflowWindow.aetherflowConsumeActions.map((action, index) => <Grid.Column key={index} width={5}>
+									<Grid.Row>
+										<ActionLink {...this.data.getAction(action.actionId)}/>
+									</Grid.Row>
+									<Grid.Row>
+										<Button
+											circular
+											compact
+											size="mini"
+											icon="time"
+											onClick={() => this.scrollToAetherflowTimeline(aetherflowWindow.timestamp)}
+										/>
+										{this.parser.formatEpochTimestamp(action.timestamp)}
+									</Grid.Row>
+								</Grid.Column>)}
+							</Grid>
+						</Table.Cell>
+						<Table.Cell>{this.AETHERFLOW_CHARGES_PER_CAST - aetherflowWindow.aetherflowConsumeActions.length || '-'}</Table.Cell>
+					</Table.Row>
+				})}
+			</Table.Body>
+		</Table></>
+	}
+
+	private onGenerateAetherflow(event: Events['action']) {
+		this.aetherflowWindows.push({
+			aetherflowGenerateActionId: event.action,
+			timestamp: event.timestamp,
+			aetherflowConsumeActions: [],
+			// Initialize to 0 since drift will be calculated in onComplete
+			drift: 0,
+		})
+	}
+
+	private onConsumeAetherflow(event: Events['action']) {
+		// If recitation is inactive, or if the aetherflow action is not a recitation action add the action to the aetherflow window
+		if (!this.recitationActive || !this.RECITATION_ACTIONS.some(recitationActionId => recitationActionId === event.action)) {
+			this.aetherflowWindows[this.aetherflowWindows.length - 1].aetherflowConsumeActions.push({
+				actionId: event.action,
+				timestamp: event.timestamp,
+			})
+			const newTotalCasts = (this.totalCastsByConsumeAction.get(event.action) ?? 0) + 1
+			this.totalCastsByConsumeAction.set(event.action, newTotalCasts)
+			this.totalAetherflowConsumeActions++
+		}
+	}
+
+	private recitationApplied() {
+		this.recitationActive = true
+	}
+
+	private recitationRemoved() {
+		this.recitationActive = false
+	}
+
+	private onComplete() {
+		// Calculate Aetherflow drift
+		const aetherflowWindows = this.aetherflowWindows
+			.filter(window => window.aetherflowGenerateActionId === this.data.actions.AETHERFLOW.id)
+		aetherflowWindows.forEach((currentWindow, index) => {
+			const previousWindow = aetherflowWindows[index-1]
+			if (!previousWindow) {
+				currentWindow.drift = 0
+			} else {
+				currentWindow.drift = this.calculateDrift(previousWindow.timestamp, currentWindow.timestamp)
+				this.aetherflowCumulativeDrift += currentWindow.drift
+			}
+		})
+
+		// Calculate Dissipation drift
+		const dissipationWindows = this.aetherflowWindows
+			.filter(window => window.aetherflowGenerateActionId === this.data.actions.DISSIPATION.id)
+		dissipationWindows.forEach((currentWindow, index) => {
+			const previousWindow = dissipationWindows[index-1]
+			if (!previousWindow) {
+				currentWindow.drift = 0
+			} else {
+				currentWindow.drift = this.calculateDrift(previousWindow.timestamp, currentWindow.timestamp)
```

Comment:
Since dissipation's cooldown is longer than aetherflow's, this is going to result in an incorrect drift calculation for dissipation.  Maybe change the calculateDrift function to expect a "skillCooldown" parameter and pass in as appropriate?

Reply:
Yeah, I whiffed on that one. Forgot to double check the dissipation drift times. Good catch.